src/main/java/org/apache/commons/imaging/formats/pnm/PnmImageParser.java

```diff
@@ -334,6 +325,12 @@ public void writeImage(final BufferedImage src, final OutputStream os, Map<Strin
                     writer = new PamWriter();
                 }
             }
+            
+            // read parameters specific for the PNM format
+            if (params instanceof ImagingParametersPnm) {
```

Comment:
Would be nice if we could use generics, instead of checking instance types. The parameters of `PnmImageParser` must definitely be an `ImagingParametersPnm` I believe.